jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/GlobalTimerService.java

```diff
@@ -372,6 +372,25 @@ public String getUuid() {
             return null;
         }
 
+        @Override
+        public int hashCode() {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + (int) (this.getId() ^ (this.getId() >>> 32));
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if ( this == obj ) return true;
+            if ( obj == null ) return false;
+            if ( getClass() != obj.getClass() ) return false;
+            final DefaultJobHandle other = (DefaultJobHandle) obj;
+            if (this.getId() != other.getId())
+                return false;
+            return true;
```

Comment:
```suggestion
          return getId() != ((DefaultJobHandle)obj).getId();
```

Reply:
SonarCloud marks it as a "code smell", because it's better not to use the if-then clause, but the boolean result to be returned as indicated by @fjtirado .

Reply:
@fjtirado thanks - I simplified the method based on your input